src/workqueue.cpp

```diff
@@ -140,33 +142,40 @@ void WorkQueue::ThreadRun()
         assert(command != nullptr);
         assert(command->has_work());
 
-        // #TODO: Spawn jobs for command to unclog the queue.
-
-        WorkQueueResultPtr result = command->work();
+        // Submit work to thread pool
+        WorkQueueResultPtr result = ThreadPoolSingleton::get_workqueue_pool()
+                                        .submit_task([cmd = std::move(command)]() { return cmd->work(); })
+                                        .get();
 
         m_lastFinishedCommandId = command->command_id;
 
-        const bool has_callback = command->has_callback();
-        const bool has_delayed_command = command->has_delayed_command();
-
         // Command work functions do not need to return a result,
         // but when using a callback or delayed command then a result is required.
-        if (result == nullptr && (has_callback || has_delayed_command))
+        if (result == nullptr)
         {
-            result = std::make_unique<WorkQueueResult>();
+            const bool has_callback = command->has_callback();
```

Comment:
Moved these consts here because command hasn't moved yet in this case. command->has_callback vs result->command->has_callback, I think it's safer this way